### PR TITLE
Update logstash-verifier image

### PR DESCRIPTION
**Pull request by msheiny**
So this PR does two things:
* updates the logstash-verifier image
* anddd updates a little bit of build logic

On the first issue, re-did the image completely based on an upstream openjdk image (MUCH easier build process with grsec), and did some re-engineering to try and make the image smaller (like grab upstream release package instead of compiling from source). I also tweaked the build entrypoint script so the image can be run as `logstash` user (🎉  without `root` ✨  ). 

So I also made some possible controversial changes to the build scripts. @conorsch need a ping here. FYI i've already built and pushed image `quay.io/freedomofpress/logstash-verifier:6.1.2` up and recorded that digest hash into the logstash-filters repo.

### CODE REVIEW

File: logstash-verifier/Dockerfile [hunk 1]
```diff
@@ -1,55 +1,35 @@
...
+    apt-get update && apt-get install logstash=1:${LS_VER}-1 && \
+    rm -rf /var/lib/apt/lists/* && \
+    rm -rf /var/cache/apt/*
+
+RUN cd /tmp && \
+    curl -sL $LS_VERIFIER_TAR -o ls-verifier.tar.gz && \
+    echo "$LS_VERIFIER_SHA256 ls-verifier.tar.gz" | sha256sum -c - && \
+    tar -C /usr/bin -xzf ls-verifier.tar.gz
+
+RUN chown logstash /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java && \
```
Comment (conorsch) [inline logstash-verifier/Dockerfile]: Wait, why does the `logstash` user need to _own_ the Java binary? It should be 755 by default.

**Review by conorsch — CHANGES_REQUESTED** [review]
Build logic overhaul looks sound! Requesting clarification on a java `chown` that I'm not convinced needs to be there.

File: logstash-verifier/Dockerfile (hunk 1 above)
Comment (msheiny) [reply logstash-verifier/Dockerfile]: Good question! Becauussee i need to change the pax flags on the java binary prior to logstash execution. In order to do that, the `logstash` user needs write permissions on it.

You might be saying... well wait a second.. i thought you already changed the pax flags earlier in the dockerfile using root permissions!? You are correct -- but unfortunately however docker handles the filesystem doesnt retain these filesystem flags and they effectively get wiped upon a new docker run.

tl;dr - in order to run the container as non-root, we need pax flags set at docker runtime by the logstash user

Comment (conorsch) [reply logstash-verifier/Dockerfile]: Thanks for clarifying, @msheiny, both here and in the appended comment. Let's dig more deeply on that separately, e.g. playing with `paxctl -c` versus `paxctl -C`.

**Review by conorsch — APPROVED** [review]
